File: snippets/dart_snippets/lib/send_spontaneous_payment.dart
import 'package:breez_sdk/breez_sdk.dart';
import 'package:breez_sdk/bridge_generated.dart';
import 'dart:convert';

Future<SendPaymentResponse> sendSpontaneousPayment({
  required String nodeId,
}) async {
  // ANCHOR: send-spontaneous-payment
  String optionalLabel = "<label>";

  SendSpontaneousPaymentRequest req = SendSpontaneousPaymentRequest(
    amountMsat: 3000000,
    nodeId: nodeId,
    label: optionalLabel,
  );
  SendPaymentResponse resp = await BreezSDK().sendSpontaneousPayment(req: req);
  // ANCHOR_END: send-spontaneous-payment
  return resp;
}

Future<SendPaymentResponse> sendSpontaneousPaymentWithTlvs({
  required String nodeId,
}) async {
  // ANCHOR: send-spontaneous-payment-with-tlvs
  List<TlvEntry> optionalExtraTlvs = [
    TlvEntry(
      fieldNumber: 34349334,
      value: utf8.encode("Hello world!"),
    )
  ];
  SendSpontaneousPaymentRequest req = SendSpontaneousPaymentRequest(
    amountMsat: 3000000,
    nodeId: nodeId,
    extraTlvs: optionalExtraTlvs,
  );
  SendPaymentResponse resp = await BreezSDK().sendSpontaneousPayment(req: req);
  // ANCHOR_END: send-spontaneous-payment-with-tlvs
  return resp;
}

File: snippets/react-native/send_spontaneous_payment.ts
import {
  sendSpontaneousPayment,
  type TlvEntry
} from '@breeztech/react-native-breez-sdk'

const exampleSendSpontaneousPayment = async () => {
  // ANCHOR: send-spontaneous-payment
  try {
    const nodeId = '...'
    const optionalLabel = '<label>'

    const sendPaymentResponse = await sendSpontaneousPayment({
      nodeId,
      amountMsat: 3000000,
      label: optionalLabel
    })
  } catch (err) {
    console.error(err)
  }
  // ANCHOR_END: send-spontaneous-payment
}

const stringToBytes = (str: string): number[] => {
  const bytes: number[] = []
  for (let i = 0; i < str.length; ++i) {
    bytes.push(str.charCodeAt(i))
  }
  return bytes
}

const exampleSendSpontaneousPaymentWithTlvs = async () => {
  // ANCHOR: send-spontaneous-payment-with-tlvs
  try {
    const nodeId = '...'
    const optionalExtraTlvs: TlvEntry[] = [
      { fieldNumber: 34349334, value: stringToBytes('Hello world!') }
    ]

    const sendPaymentResponse = await sendSpontaneousPayment({
      nodeId,
      amountMsat: 3000000,
      extraTlvs: optionalExtraTlvs
    })
  } catch (err) {
    console.error(err)
  }
  // ANCHOR_END: send-spontaneous-payment-with-tlvs
}
